app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+import math
+
+
 class Vector:
-    # write your code here
-    pass
+    def __init__(self, x: int, y: int) -> None:
+        self.x = round(x, 2)
+        self.y = round(y, 2)
+
+    def __add__(self, vector_2: Vector) -> Vector:
+        return Vector(self.x + vector_2.x, self.y + vector_2.y)
+
+    def __sub__(self, vector_2: Vector) -> Vector:
+        return Vector(self.x - vector_2.x, self.y - vector_2.y)
+
+    def __mul__(self, multiplier: Vector | int) -> float | Vector:
+        if isinstance(multiplier, Vector):
+            return self.x * multiplier.x + self.y * multiplier.y
+
+        elif isinstance(multiplier, int) or isinstance(multiplier, float):
+            return Vector(self.x * multiplier, self.y * multiplier)
```

Comment:
Można to trochę uprościć
```suggestion
    if isinstance(other, (int, float)): 
            return Vector(self.x * other, self.y * other)
    return self.x * other.x + self.y * other.y 
```